src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,94 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final int MULTIPLIER = 2;
+    private static final float LOAD_FACTOR = 0.75f;
+    private int threshold;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        threshold = (int) (DEFAULT_CAPACITY * LOAD_FACTOR);
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = getIndex(key);
+        Node<K, V> newNode = new Node<>(Objects.hash(key), key, value, null);
+        if (table[index] == null) {
+            table[index] = newNode;
+        } else {
+            Node<K, V> currentBucket = table[index];
+            while (currentBucket != null) {
+                if (Objects.equals(currentBucket.key, key)) {
+                    currentBucket.value = value;
```

Comment:
Not a big deal, just a context
A cell int a table is a Bucket, so you're iterating on nodes not buckets